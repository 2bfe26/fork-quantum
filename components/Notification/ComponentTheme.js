import Colors from '../Colors';

const ComponentTheme = {
  default: {
    backgroundColor: Colors.SECONDARY['500'],
    borderColor: Colors.SECONDARY['600'],
    color: Colors.WHITE,
  },
  success: {
    backgroundColor: Colors.ACTION['400'],
    borderColor: Colors.ACTION['600'],
    color: Colors.WHITE,
  },
  warning: {
    backgroundColor: Colors.WARNING['400'],
    borderColor: Colors.WARNING['600'],
    color: Colors.WHITE,
  },
  error: {
    backgroundColor: Colors.DANGER['400'],
    borderColor: Colors.DANGER['600'],
    color: Colors.WHITE,
  },
  info: {
    backgroundColor: Colors.INFO['100'],
    borderColor: Colors.INFO['300'],
    color: Colors.WHITE,
  },
};

export default ComponentTheme;
