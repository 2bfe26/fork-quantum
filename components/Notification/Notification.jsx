import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import React from 'react';

import GlobalTheme from '../../theme';
import ComponentTheme from './ComponentTheme';

/** A notification component that alerts user of something */
class Notification extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      show: this.props.show,
    };

    if (this.props.duration) {
      this.handleAutoDestruction();
    }
  }

  hideModal = () => {
    this.setState({ show: false }, this.props.onClose ? this.props.onClose() : null);
  }

  handleAutoDestruction = () => {
    setTimeout(() => {
      this.hideModal();
    }, this.props.duration * 1000);
  }

  render() {
    const { message: { title, text, type } } = this.props;

    const notification = this.state.show ? (
      <ThemeProvider theme={GlobalTheme}>
        <NotificationWrapper type={type}>
          <NotificationIcon />

          <NotificationMessage>
            <strong>
              {title}
            </strong>

            {text}
          </NotificationMessage>

          <NotificationClose onClick={this.hideModal}><span aria-hidden="true">Ã—</span></NotificationClose>
        </NotificationWrapper>
      </ThemeProvider>
    ) : null;

    return notification;
  }
}

const NotificationWrapper = styled.div`
  ${props => ComponentTheme[props.type]}
  padding: ${props => props.theme.sizes.spacing};
  border-radius: ${props => props.theme.sizes.radius};
  position: relative;
`;

const NotificationIcon = styled.div`
`;

const NotificationMessage = styled.div`
  strong {
    display: ${props => (props.inline ? 'block' : 'inline-block')};
    margin-right: 5px;
  }
  margin-right: 15px;
`;

const NotificationClose = styled.button`
    position: absolute;
    cursor: pointer;
    top: 12px;
    right: 0;
    padding: 0.75rem 1.25rem;
    color: inherit;
    background-color: transparent;
    border: 0;
    -webkit-appearance: none;
`;

Notification.defaultProps = {
  show: true,
  duration: 0,
  message: {
    title: 'Sample', text: 'This is a dummy', type: 'info',
  },
  inline: true,
  onClose: () => {},
};

Notification.propTypes = {
  /** it says to component if it should appear or not */
  show: PropTypes.bool,
  /** time in seconds that notification is visible */
  duration: PropTypes.number,
  /** object with message details */
  message: PropTypes.shape({
    /** message title */
    title: PropTypes.string,
    /** message text */
    text: PropTypes.string,
    /** message type */
    type: PropTypes.oneOf([
      'default',
      'success',
      'warning',
      'error',
      'info',
    ]),
  }),
  /** it says if the title should stay in the same line of the message itself */
  inline: PropTypes.bool, // eslint-disable-line react/no-unused-prop-types
  /** callback function */
  onClose: PropTypes.func,
};

export default Notification;
