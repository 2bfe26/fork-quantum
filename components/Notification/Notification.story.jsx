import React from 'react';
import { storiesOf } from '@storybook/react';
import Heading from '../../.storybook/components/Heading';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import ComponentPanel from '../../.storybook/components/ComponentPanel';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import Notification from './Notification';

storiesOf('4. Notifications', module)
  .add('Basic', () => (
    <Heading atom title="Notification">
      <TabbedView>
        <Tab title="Usage">
          <ComponentPanel component={<Notification />} importModules="Notification" />
        </Tab>

        <Tab title="API">
          <AutoPropsApi component={Notification} />
        </Tab>
      </TabbedView>
    </Heading>
  ));
