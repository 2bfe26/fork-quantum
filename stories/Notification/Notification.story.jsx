import React from 'react';
import { storiesOf } from '@storybook/react';
import Heading from '../../.storybook/components/Heading';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import ComponentPanel from '../../.storybook/components/ComponentPanel';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import Notification from '../../components/Notification';
import Atom from '../static/atom.svg';

storiesOf('4. Notifications', module)
  .add('Basic', () => (
    <Heading image={Atom} title="<Notification />">
      <TabbedView>
        <Tab title="Usage">
          <ComponentPanel component={<Notification />} importModules="Notification" />
        </Tab>

        <Tab title="API">
          <AutoPropsApi component={Notification} />
        </Tab>
      </TabbedView>
    </Heading>
  ));
