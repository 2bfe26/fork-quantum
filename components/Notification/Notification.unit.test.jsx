import React from 'react';
import renderer from 'react-test-renderer';
import Notification from './Notification';

describe('Input component ', () => {
  it('should match the snapshot', () => {
    expect(renderer.create(<Notification />).toJSON()).toMatchSnapshot();
  });
});
