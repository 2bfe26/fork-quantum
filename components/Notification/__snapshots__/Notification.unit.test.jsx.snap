// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component  should match the snapshot 1`] = `
<div
  className="sc-bdVaJa fRinFw"
  type="info"
>
  <div
    className="sc-bwzfXH bFHdpU"
  />
  <div
    className="sc-htpNat ecwypm"
  >
    <strong>
      Sample
    </strong>
    This is a dummy
  </div>
  <button
    className="sc-bxivhb fRngHK"
    onClick={[Function]}
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
